### ISSUE TITLE
Feature request: Fixing "expected 1 argument, found 2" by adding new argument

### ISSUE THREAD

**Issue by vi** [rust-analyzer/rust-analyzer#11642]
Sometimes I want to initiate adding new argument to a function when the editor is positioned at call site, not at definition site. I want Rust Analyzer to be able to update function definition with a new argument when I it encounters `this function takes 1 argument but 2 arguments were supplied` error.

Before:

```rust
fn foo(){}
fn bar() {
    let x : i32 = 5;
    foo(x┃);
}
```

After:

```rust
fn foo(x: i32){}
fn bar() {
    let x : i32 = 5;
    foo(x┃);
}
```

It should also support modifying trait methods.

It is questionable whether should also automatically find other call sites and insert `todo!()`s there. Maybe this should be triggerable by a separate code assist.

**Comment by flodiebold** [comment]
Duplicate of #6499.
